### PR TITLE
User dashboard

### PR DESCRIPTION
**Pull request by dev-enforced**
This pull request consists of all the components required to develop the User Dashboard page which the user will get to see once the name of the user is entered:
1) Weather Component at the top right that provides information about the weather details of the area
2) Clock Component at the center that tells the user about the current time and what period of the day it is (morning ,evening, afternoon or night depending on the time). User has the option to toggle the time format by clicking on the time
3) A goal component that displays the user's current goal or gives an option to set the goal if the user is using the extension for first time along with option to edit / delete the current set goal
4) Settings component at the  bottom left enabling the user to change the name redirecting him to the user onboarding page that requests the user's new name to get back to the dashboard
5) A quotes component that displays the quote and the author at the bottom center 

This page isn't responsive as it is a browser extension so focussed to work only on big screens such as laptop and desktops
Live preview Link: https://polo-extension-git-user-dashboard-dev-enforced.vercel.app/

### CODE REVIEW

**Review by AtulPant2704 — APPROVED** [review]
- Add `cursor: pointer` to settings icon
- Don't add `required` to task input, everytime user comes to home screen a popup comes over there Please fill out this field which is not a good UX.
- Rest all looks fine.

File: src/components/GetFocus/GetFocus.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    if (localStorage.getItem("currentTaskState") !== null) {
+      return JSON.parse(localStorage.getItem("currentTaskState") || "");
+    } else {
+      return defaultGoalState;
+    }
+  });
+  const submitHandler = (submitEvent: FormEvent) => {
+    submitEvent.preventDefault();
+  };
+  const receiveGoal = (keyAction: React.KeyboardEvent<HTMLInputElement>) => {
```
Comment (ankit9015) [inline src/components/GetFocus/GetFocus.tsx:21]: I like that you are using keyboard events in this

File: src/components/GetTimeAndDate/GetTimeAndDate.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const hoursAndMinutesIn12 = `${hoursIn12}:${minutesFormatted}`;
+  const periodOfTheDayInfo = getPeriodOfTheDay(hours);
+  useEffect(() => {
+    let timer = setTimeout(() => {
+      updateTimeAndDateDetails(() => new Date());
+    }, 1000);
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [timeAndDateDetails]);
```
Comment (ankit9015) [inline src/components/GetTimeAndDate/GetTimeAndDate.tsx:25]: Check dependency array once

**Review by ankit9015 — APPROVED** [review]
Code is well structured and easy to understand.

**Comment by Hardiegogo** [comment]
Extension looks cool.
State persists after refreshing which is good.
Good work Roshan.

File: src/components/GetFocus/GetFocus.tsx (hunk 1 above)
Comment (dev-enforced) [reply src/components/GetFocus/GetFocus.tsx:21]: Yes Typescript required me to provide details of the event so accessed the event provided from the React event setup

**Pull request merged**